app/gateways/http/authentication/login_test.go

```diff
@@ -0,0 +1,173 @@
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader(requestBody))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.Login).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected, _ := json.Marshal(responseBody)
+
+		assert.Equal(t, http.StatusCreated, response.Code)
+		assert.Equal(t, string(expected), strings.TrimSpace(got))
+		assert.Equal(t, http_server.JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when body has invalid fields", func(t *testing.T) {
+		body := LoginBadRequest{
+			cpf:    12345678910,
+			secret: "secret",
+		}
+		requestBody, _ := json.Marshal(body)
+		err := errors.New("invalid fields")
+		handler := createFakeHandler("", err, nil)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader(requestBody))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.Login).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid fields"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, http_server.JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when body is empty", func(t *testing.T) {
+		body := LoginRequest{
+			CPF:    "",
+			Secret: "",
+		}
+		requestBody, _ := json.Marshal(body)
+		err := errors.New("invalid fields")
+		handler := createFakeHandler("", nil, err)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader(requestBody))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.Login).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid fields"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, http_server.JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when is missing fields", func(t *testing.T) {
+		err := errors.New("invalid request payload")
+		handler := createFakeHandler("", nil, err)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader([]byte{}))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.Login).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid request payload"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, http_server.JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when CPF does not exist", func(t *testing.T) {
+		body := LoginRequest{
+			CPF:    "12345678913",
+			Secret: "MySecret",
+		}
+		requestBody, _ := json.Marshal(body)
+		err := errors.New("incorrect username or password")
+		handler := createFakeHandler("", nil, err)
```

Comment:
I believe the fake handler is being created in the wrong way. since you become aware that a CPF doesn't exist by calling GetByCPF, but you're mocking CreateToken now. Do you agree?